src/utils/types.ts

```diff
@@ -1,5 +1,14 @@
 // This file contains common types and interfaces used across the application.
 
+// Define ToolAnnotations interface based on MCP schema 2025-03-26
+export interface ToolAnnotations {
+  destructiveHint?: boolean;
+  idempotentHint?: boolean;
+  openWorldHint?: boolean;
+  readOnlyHint?: boolean;
+  title?: string;
+}
+
 // Define Tool interface
```

Comment:
I think there is a reusable type we could use here instead since it's in the MCP SDK. Benefit is that it will have all of the jsdoc.

Reply:
https://github.com/modelcontextprotocol/typescript-sdk/blob/7e18c7016943af389b16bd670671c3527e3da21d/src/types.ts#L1310

Reply:
Looks like this is another package.